### PR TITLE
Dispatcher to center floating window

### PR DESCRIPTION
**Pull request by abbadanor**
I had some trouble using hyprctl to center my window, doing things like `hyprctl dispatch movewindowpixel 50% 50% active`, which did not work. If there is some way to achieve this with the current version, then please show how.

With this PR, it's as easy as `hyprctl dispatch centerwindow active`

If this PR is accepted, I will make a PR in the wiki.

Btw, although the code is working, I don't know C++, so I can't guarantee the stability/quality of my code.

### CODE REVIEW

File: src/managers/KeybindManager.cpp [hunk 1]
```diff
@@ -579,6 +580,19 @@ void CKeybindManager::toggleActiveFloating(std::string args) {
     g_pLayoutManager->getCurrentLayout()->changeWindowFloatingMode(PWINDOW);
 }
 
+void CKeybindManager::centerWindow(std::string args) {
+    CWindow* PWINDOW = nullptr;
+
+    PWINDOW = g_pCompositor->m_pLastWindow;
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:586]: why is this two lines? also, PWINDOW can be made const. That's what ALLCAPS means.

File: src/managers/KeybindManager.cpp [hunk 2]
```diff
@@ -579,6 +580,19 @@ void CKeybindManager::toggleActiveFloating(std::string args) {
...
+
+    if (!PWINDOW || !PWINDOW->m_bIsFloating || PWINDOW->m_bIsFullscreen)
+        return;
+
+    const auto PMONITOR = g_pCompositor->getMonitorFromID(PWINDOW->m_iMonitorID);
+
+    PWINDOW->m_vRealPosition = PMONITOR->vecPosition + PMONITOR->vecSize / 2.f - PWINDOW->m_vRealSize.goalv() / 2.f;
```
Comment (vaxerski) [inline src/managers/KeybindManager.cpp:593]: for floating windows, also update PWINDOW->m_vPosition

Comment (abbadanor) [reply src/managers/KeybindManager.cpp:593]: Ok wait a sec

**Review by abbadanor — COMMENTED** [review]
it now also updates vPosition

**Pull request merged**